modules/custom/az_news/az_news_feeds/src/Form/AzNewsFeedsAdminForm.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+namespace Drupal\az_news_feeds\Form;
+
+use Drupal\Core\Form\ConfigFormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ *
+ */
+class AzNewsFeedsAdminForm extends ConfigFormBase {
+
+  /**
+   *
+   */
+  protected function getEditableConfigNames() {
+    return [
+      'migrate_plus.migration_group.az_news_feeds',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'az_news_feeds_admin';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $config = $this->config('migrate_plus.migration_group.az_news_feeds');
+    $form['urls'] = [
+      '#type' => 'textarea',
+      '#title' => $this->t('UArizona News Feed URLs'),
+      '#description' => $this->t('URLs to fetch.'),
+      '#default_value' => $config->get('shared_configuration.source.urls'),
+    ];
+
+    return parent::buildForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    parent::submitForm($form, $form_state);
+
+    $this->config('migrate_plus.migration_group.az_news_feeds')
+      ->set('shared_configuration.source.urls', $form_state->getValue('urls'))
+      ->save();
+  }
```

Comment:
I think we can add an event dispatcher here.
https://git.drupalcode.org/project/drupal/-/blob/9.3.x/core/modules/migrate/migrate.services.yml#L2

Docs: https://api.drupal.org/api/drupal/core%21modules%21migrate%21src%21Plugin%21PluginEventSubscriber.php/function/PluginEventSubscriber%3A%3ApreImport/8.2.x

https://www.computerminds.co.uk/drupal-code/drupal-8-event-subscribers-successor-alter-hooks